### PR TITLE
Ignoring error if more than max amount of touches present

### PR DESCRIPTION
**Pull request by jimon**
### Description

- If `HasButtonPress` is called on incoming events.
- And more than 10 touches are present.
- Then `GetStateOffsetForEvent(null, ...)` returns false for touchscreen, because it can't find a free slot for incoming touch.
- That leads to `InputEventControlEnumerator.Reset()` throws an exception.

### Changes made

Ignoring the exception for specific case of touch events to touchscreen.

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1506) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/InputSystem?pullRequest=1506) before we can accept your contribution.<br/><hr/>**Dmytro Ivanov** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/InputSystem?pullRequest=1506) it.</sub>

File: Assets/Tests/InputSystem/CoreTests_Devices.cs [hunk 1]
```diff
@@ -5480,4 +5480,20 @@ public void Devices_EventPreProcessor_Can(PreProcessorTestDevice.Behavior second
...
+    {
+        var touch = InputSystem.AddDevice<Touchscreen>();
+
+        InputSystem.onEvent += (InputEventPtr eventPtr, InputDevice device) =>
+        {
+            eventPtr.HasButtonPress();
+        };
+
+        for (var i = 0; i < TouchscreenState.MaxTouches + 5; ++i)
+            BeginTouch(i, new Vector2(i * 1.0f, i * 2.0f), time: 0);
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Devices.cs]: Maybe

```
Assert.That(()=>{
        for (var i = 0; i < TouchscreenState.MaxTouches + 5; ++i)
            BeginTouch(i, new Vector2(i * 1.0f, i * 2.0f), time: 0); }, Does.Not.Throw);
```

just for the sake of being super explicit.

**Review by Rene-Damm — APPROVED** [review]
Unfortunate. But also don't see a better way without applying a deeper level change.

File: Assets/Tests/InputSystem/CoreTests_Devices.cs [hunk 2]
```diff
@@ -5480,4 +5480,20 @@ public void Devices_EventPreProcessor_Can(PreProcessorTestDevice.Behavior second
...
+    // https://fogbugz.unity3d.com/f/cases/1395648/
+    [Test]
+    [Category("Devices")]
+    public unsafe void Devices_DoesntErrorOutOnMaxTouchCount()
+    {
+        var touch = InputSystem.AddDevice<Touchscreen>();
+
+        InputSystem.onEvent += (InputEventPtr eventPtr, InputDevice device) =>
+        {
+            eventPtr.HasButtonPress();
```
Comment (Rene-Damm) [inline Assets/Tests/InputSystem/CoreTests_Devices.cs]: Or maybe even put the assert here.

```
Assert.That(() => eventPtr.HasButtonPress(), Does.Not.Throw());

Comment (jimon) [reply Assets/Tests/InputSystem/CoreTests_Devices.cs]: Guess it's `Throws.Nothing` :) Fixed

File: Assets/Tests/InputSystem/CoreTests_Devices.cs (hunk 1 above)
Comment (jimon) [reply Assets/Tests/InputSystem/CoreTests_Devices.cs]: Fixed

**Pull request merged**